src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -697,12 +733,13 @@ protected internal void OnDocumentsAdded(ImmutableArray<DocumentInfo> documentIn
         {
             using (_serializationLock.DisposableWait())
             {
-                var oldSolution = this.CurrentSolution;
-                var newSolution = this.SetCurrentSolution(oldSolution.AddDocuments(documentInfos));
+                var (oldSolution, newSolution) = this.SetCurrentSolutionEx(this.CurrentSolution.AddDocuments(documentInfos));
+                var projectIds = documentInfos.Select(i => i.Id.ProjectId).Distinct();
 
                 // Raise ProjectChanged as the event type here. DocumentAdded is presumed by many callers to have a
                 // DocumentId associated with it, and we don't want to be raising multiple events.
-                this.RaiseWorkspaceChangedEventAsync(WorkspaceChangeKind.ProjectChanged, oldSolution, newSolution);
+                foreach (var projectId in projectIds)
+                    this.RaiseWorkspaceChangedEventAsync(WorkspaceChangeKind.ProjectChanged, oldSolution, newSolution, projectId);
```

Comment:
this looks to be a bug fix here.  We were issuing ProjectChanged without any actual projectIds.

this can't current use SetCurentSolution(transform) because it issues multiple events now.  We could have a generic overload of SetCurrentSolution that just handles all the 'after work' and thus could have this loop in there.  lmk if you'd like that @jasonmalinowski .